Makefile

```diff
@@ -284,6 +288,51 @@ bin/releases/git-lfs-%-$(VERSION).zip : $(RELEASE_INCLUDES) bin/git-lfs-%.exe
 bin/releases/git-lfs-$(VERSION).tar.gz :
 	git archive -o $@ --prefix=git-lfs-$(patsubst v%,%,$(VERSION))/ --format tar.gz $(VERSION)
 
+# release-windows is a target that builds and signs Windows binaries.  It must
+# be run on a Windows machine under Git Bash.
+#
+# You may sign with a different certificate by specifying CERT_SHA1.
+.PHONY : release-windows
+release-windows: bin/releases/git-windows-assets-$(VERSION).tar.gz
+
+bin/releases/git-windows-assets-$(VERSION).tar.gz :
+	$(RM) git-lfs-windows-*.exe
+	@# Using these particular filenames is required for the Inno Setup script to
+	@# work properly.
+	$(MAKE) -B GOARCH=amd64 && cp ./bin/git-lfs.exe ./git-lfs-x64.exe
+	$(MAKE) -B GOARCH=386 && cp ./bin/git-lfs.exe ./git-lfs-x86.exe
+	signtool.exe sign /sha1 $(CERT_SHA1) /fd sha256 /tr http://timestamp.digicert.com /td sha256 /v git-lfs-x64.exe
+	signtool.exe sign /sha1 $(CERT_SHA1) /fd sha256 /tr http://timestamp.digicert.com /td sha256 /v git-lfs-x86.exe
+	iscc.exe script/windows-installer/inno-setup-git-lfs-installer.iss
+	@# This file will be named according to the version number in the
+	@# versioninfo.json, not according to $(VERSION).
+	mv git-lfs-windows-*.exe git-lfs-windows.exe
+	signtool.exe sign /sha1 $(CERT_SHA1) /fd sha256 /tr http://timestamp.digicert.com /td sha256 /v git-lfs-windows.exe
+	mv git-lfs-x64.exe git-lfs-windows-amd64.exe
+	mv git-lfs-x86.exe git-lfs-windows-386.exe
+	@# We use tar because Git Bash doesn't include zip.
+	tar -cf $@ git-lfs-windows-amd64.exe git-lfs-windows-386.exe git-lfs-windows.exe
+	$(RM) git-lfs-windows-amd64.exe git-lfs-windows-386.exe git-lfs-windows.exe
+
+# release-windows-rebuild takes the archive produced by release-windows and
+# incorporates the signed binaries into the existing zip archives.
+.PHONY : release-windows-rebuild
+release-windows-rebuild:
+	[ -f "bin/releases/git-windows-assets-$(VERSION).tar.gz" ] || { \
+		echo "Windows assets are missing." >&2; exit 1; }
+	temp=$$(mktemp -d); \
+	file="$$PWD/bin/releases/git-windows-assets-$(VERSION).tar.gz"; \
+		( \
+			tar -C "$$temp" -xzf "$$file" && \
+			for i in amd64 386; \
```

Comment:
I think that flattening these is fine. It might be preferable to have a flat structure like the following, instead of the increased level in indentation. These lines look similarly enough that I imagine a reader could quickly spot the difference in `amd64` vs `386`.

```Makefile
( \
        zip -j -l bin/releases/git-lfs-windows-$(VERSION).zip "$$temp/git-lfs-windows-amd64.exe"; \
        zip -j -l bin/releases/git-lfs-windows-$(VERSION).zip "$$temp/git-lfs-windows-386.exe"; \
        cp "$$temp/git-lfs-windows.exe" bin/releases/git-lfs-windows-$(VERSION).exe; \
\ )
```

Reply:
Okay, will revise.

Reply:
You actually pointed out a bug here, which is that the zip file should have the architecture in its name.

Reply:
I actually found several issues with this on a second glance. Because the code now uses the variables multiple times in a multi-command loop, I think it's better to keep it as a loop.